app/Http/Controllers/Table/EventlogController.php

```diff
@@ -61,7 +62,15 @@ protected function filterFields($request)
      */
     public function baseQuery($request)
     {
-        return Eventlog::hasAccess($request->user())->with('device');
+        $query = Eventlog::hasAccess($request->user())->with('device');
+
+        if ($request->device_group) {
+            $query->whereHas('device.groups', function ($query) use ($request) {
```

Comment:
Should probably make this a query scope on the Device model (like hasAccess)

Reply:
True, will fix.

Reply:
That would also make it so you don't have to use all those clones below.  Those aren't very nice to look at :)